lib/components/joystick/Joystick_directional.dart

```diff
@@ -0,0 +1,286 @@
+import 'dart:math';
+
+import 'package:flame/components/joystick/joystick_component.dart';
+import 'package:flame/components/joystick/joystick_events.dart';
+import 'package:flame/gestures.dart';
+import 'package:flame/sprite.dart';
+import 'package:flutter/material.dart';
+
+class JoystickDirectional {
+  static const double _backgroundAspectRatio = 2.2;
+
+  final double size;
+  final Sprite spriteBackgroundDirectional;
+  final Sprite spriteKnobDirectional;
+  final bool isFixed;
+  final EdgeInsets margin;
+  final Color color;
+
+  Paint _paintBackground;
+  Paint _paintKnob;
+
+  Sprite _backgroundSprite;
+  Sprite _knobSprite;
+
+  Rect _backgroundRect;
+  Rect _knobRect;
+
+  bool _dragging = false;
+  Offset _dragPosition;
+  double _tileSize;
+
+  JoystickController _joystickController;
+
+  Size _screenSize;
+
+  DragEvent _currentDragEvent;
+
+  JoystickDirectional({
+    this.spriteBackgroundDirectional,
+    this.spriteKnobDirectional,
+    this.isFixed = true,
+    this.margin = const EdgeInsets.only(left: 100, bottom: 100),
+    this.size = 80,
+    this.color = Colors.blueGrey,
+  }) {
+    if (spriteBackgroundDirectional != null) {
+      _backgroundSprite = spriteBackgroundDirectional;
+    } else {
+      _paintBackground = Paint()
+        ..color = color.withOpacity(0.5)
+        ..style = PaintingStyle.fill;
+    }
+    if (spriteKnobDirectional != null) {
+      _knobSprite = spriteKnobDirectional;
+    } else {
+      _paintKnob = Paint()
+        ..color = color.withOpacity(0.8)
```

Comment:
This opacity could be a parameter, What do you think?

Reply:
Goode idea!